src/components/carousel/index.js

```diff
@@ -107,7 +107,8 @@ export default class Carousel extends BaseComponent {
 
   onOrientationChanged = () => {
     if (!this.props.pageWidth || this.props.loop) {
-      this.setState({pageWidth: Constants.screenWidth});
+      // HACK: setting to containerWidth for Android's call when view disappear
+      this.setState({pageWidth: this.state.containerWidth || Constants.screenWidth});
```

Comment:
@Inbal-Tish I think this fix breaks iOS when changing orientation

Reply:
It shouldn't break the non-full width since there is a condition on line #111. The odd thing is that `this.props.pageWidth` is undefined there for some reason...
I see an issue with orientation change in the dynamic width feature as well. I'll take a look and talk to you.

Reply:
There was an issue with the update.initialOffset (used initialPage instead of state.currentPage). That might be the orientation issue you had? I added a fix so let me know if that worked or there was another issue in orientation.